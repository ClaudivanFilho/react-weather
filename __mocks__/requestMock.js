export function fetchRequestMock() {
  return new Promise((resolve, reject) => {
    process.nextTick(
      resolve('success')
    );
  });
}