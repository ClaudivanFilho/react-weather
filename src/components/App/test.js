import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

import { App } from './index';
import { fetchRequestMock } from '../../../__mocks__/requestMock' 

const defaultProps = {
  currentWeather: {
    weather: null,
    loading: false
  }
}

configure({ adapter: new Adapter() });

test('Test App Component', () => {
  const component = shallow(<App {...defaultProps} />);
  expect(component.state()).toEqual({
    view: 'CURRENT_WEATHER',
    position: { lat: -34.397, lng: 150.644 },
    zoom: 7
  });

  expect(toJson(component)).toMatchSnapshot();
});

test('Test App Fetch Weather Function', () => {
  const component = shallow(
    <App {...defaultProps} fetchCurrentWeather={fetchRequestMock} fetchForecast={fetchRequestMock}/>
  );
  component.instance().fetchWeather(20, -40);
  expect(component.state().position).toEqual({
    lat: 20, lng: -40
  });
})
