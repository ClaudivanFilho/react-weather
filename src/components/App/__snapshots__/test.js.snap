// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component 1`] = `
<BrowserRouter>
  <div>
    <SuperiorMenu />
    <div
      className="container"
      style={
        Object {
          "marginTop": "50px",
        }
      }
    >
      <SuperiorHeader
        weather={null}
      />
      <div
        className="col-sm-12 float-left"
      >
        <withProps(withScriptjs(withGoogleMap(Component)))
          fetchWeather={[Function]}
          position={
            Object {
              "lat": -34.397,
              "lng": 150.644,
            }
          }
          zoom={7}
        />
      </div>
      <CSSTransitionGroup
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={300}
        transitionLeave={true}
        transitionLeaveTimeout={30}
        transitionName="loading"
      >
        <Loading
          active={false}
        />
      </CSSTransitionGroup>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/forecast"
      />
    </div>
  </div>
</BrowserRouter>
`;
